### PR TITLE
Scheduler: wrap all errors in pkg/scheduler

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This one will make all current errors have been wrapped in pkg/scheduler.

I will rethink about whether it is worth to create an error type for the framework, so I will keep the issue open.

**Which issue(s) this PR fixes**:

Part of #94695

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by gavinfish** [comment]
/assign @alculquicondor 
@alculquicondor sorry to assign two PRs to you at the same time, feel free to hold any of them to reduce your review burden:)

File: pkg/scheduler/framework/plugins/noderesources/least_allocated.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (la *LeastAllocated) Name() string {
 func (la *LeastAllocated) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
 	nodeInfo, err := la.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
 	if err != nil {
-		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
+		return 0, framework.AsStatus(fmt.Errorf("failed to get node %q from Snapshot: %w", nodeName, err))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/noderesources/least_allocated.go:50]: revert wording:

"failed" is redundant, because this is an error.

**Comment by alculquicondor** [comment]
Prefer to directly squash when the PR is small

/approve

**Comment by gavinfish** [comment]
> Prefer to directly squash when the PR is small
> 
> /approve

Squashed and thanks for advice.

**Pull request merged**